src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -82,7 +82,7 @@ internal Conversions Conversions
             {
                 if (_conversions == null)
                 {
-                    Interlocked.CompareExchange(ref _conversions, new BuckStopsHereBinder(this).Conversions, null);
+                    Interlocked.CompareExchange(ref _conversions, new BuckStopsHereBinder(this, associatedSyntaxTree: null).Conversions, null);
```

Comment:
>new BuckStopsHereBinder(this, associatedSyntaxTree: null) [](http://example.com/codeflow?start=66&length=57)

This binder will persist: it's saved inside the `Conversions` instance and then it's used for some conversions.
As far as I can tell, it's used for `GetStackAllocConversion`, `GetMethodGroupDelegateConversion`, `MethodGroupConversion`, `GetMethodGroupFunctionPointerConversion`. That means those conversions will not honor `file` accessibility on certain special types, which seems acceptable loss.

Is that the only impact of not having an associated syntax tree for this binder?

Reply:
I believe so, yes.